crates/core/src/models/call.rs

```diff
@@ -2,22 +2,52 @@ use std::slice::SliceIndex;
 
 use reth_primitives::TransactionSigned;
 use reth_rlp::Decodable;
-use starknet::accounts::Call;
+use starknet::accounts::Call as StarknetCall;
 use starknet_crypto::FieldElement;
 
 use super::ConversionError;
 use crate::models::DataDecodingError;
 
-pub struct Calls(Vec<Call>);
+#[derive(Clone)]
+pub struct Call(StarknetCall);
 
-impl From<Calls> for Vec<Call> {
+impl From<Call> for StarknetCall {
+    fn from(call: Call) -> Self {
+        call.0
+    }
+}
+
+impl From<StarknetCall> for Call {
+    fn from(call: StarknetCall) -> Self {
+        Self(call)
+    }
+}
+
+impl From<Call> for Vec<FieldElement> {
+    fn from(call: Call) -> Self {
+        let mut c = vec![
+            FieldElement::ONE,
```

Comment:
Maybe we could add some comments on the structure of how a vector calldata is constructed? Got a bit confused of why the "One" and then the "Zero"